zio-http/shared/src/main/scala/zio/http/codec/internal/EncoderDecoder.scala

```diff
@@ -540,11 +535,7 @@ private[codec] object EncoderDecoder {
         if (inputs.length > 1) {
           Body.fromMultipartForm(encodeMultipartFormData(inputs, outputTypes), formBoundary)
         } else {
-          if (isEventStream) {
-            Body.fromCharSequenceStreamChunked(
-              inputs(0).asInstanceOf[ZStream[Any, Nothing, ServerSentEvent]].map(_.encode),
-            )
-          } else if (inputs.length < 1) {
+          if (inputs.length < 1) {
```

Comment:
nitpick: How about replacing the nested `if`s with pattern matching for better readability?

```scala
input.length match {
  case 0 => 
    Body.empty
  
  case 1 =>
    val bodyCodec = flattened.content(0)
    bodyCodec.erase.encodeToBody(inputs(0), outputTypes)
  
  case _ =>  
    Body.fromStreamChunked(inputs(0).asInstanceOf[ZStream[Any, Nothing, Byte]])
}
```